### DISCUSSION TITLE
multi authority end points

### DISCUSSION THREAD

**Discussion by mehdy-dev** [ThreeMammals/Ocelot#2157] [Q&A] [answered]
Hello guys, as we have several authentication provider for different end users, so i wanted to know is it possible to fallback authentication loop towards second provider or third, in the case first failed, as a work around if we create wrapper routes for same downstream and register second authentication provider in the start up it will work, while without upstream wrapper is it possible? I mean having array of token authentication provider instead of only one. 

Instead of
```json
"AuthenticationOptions": {
                "AuthenticationProviderKey": "Bearer",
                "AllowedScopes": []
            },
```
having 
```json
"AuthenticationOptions": {
                "AuthenticationProviderKey": ["Bearer","app1","app2"]
                "AllowedScopes": []
            },
```
or manage the fallback in the case of failure for second app ? 
btw this approach will increases network thruputs even if also possible, :). thanks.

**Comment by raman-m** [comment, marked as answer]
Hello, Mehdi!

Have you reviewed the [Authentication](https://ocelot.readthedocs.io/en/latest/features/authentication.html) documentation? You mentioned an authentication scenario that involves [Multiple Authentication Schemes](https://ocelot.readthedocs.io/en/latest/features/authentication.html#multiple-authentication-schemes) within a single route definition. Please ensure that all relevant authentication providers are correctly configured in the C# code.

Lastly, consider testing out this feature in Ocelot: [Multiple Authentication Schemes](https://ocelot.readthedocs.io/en/latest/features/authentication.html#multiple-authentication-schemes).
